### ISSUE TITLE
I find a practical way to Environment setup and inference on own dataset

### ISSUE THREAD

**Issue by 3406212002** [JonasSchult/Mask3D#192]
I have compiled the environment for GPUs supporting CUDA 11.3, such as those with Ampere architecture or earlier. Note that GPUs based on the Ada Lovelace architecture and beyond do not support CUDA 11.3.

________________________________________________________________________________________________________________________________________________
apt-get update
apt-get install -y libopenblas-dev
export TORCH_CUDA_ARCH_LIST="6.0 6.1 6.2 7.0 7.2 7.5 8.0 8.6"

git clone https://github.com/JonasSchult/Mask3D.git
cd Mask3D
conda create -n mask3d python=3.10.9 -y 
conda activate mask3d
 
pip3 install torch==1.12.1+cu113 torchvision==0.13.1+cu113 pytorch-lightning==1.7.2 torchmetrics==0.11.4 --extra-index-url https://download.pytorch.org/whl/cu113
pip3 install torch-scatter -f https://data.pyg.org/whl/torch-1.12.1+cu113.html
pip3 install 'git+https://github.com/facebookresearch/detectron2.git@710e7795d0eeadf9def0e7ef957eea13532e34cf'

pip install pip==24.0
pip install "cython<3.0.0"
pip install --no-build-isolation "pyyaml==5.4.1"
pip install hydra-core==1.0.5
pip install pycocotools 
pip install numpy==1.23.5 tensorboard loguru natsort fire scikit-learn scikit-image pillow==9.5.0  matplotlib pyviz3d python-dotenv plyfile trimesh wandb==0.15.0 fvcore cloudpickle albumentations==1.2.1

pip install -U git+https://github.com/kumuji/volumentations
pip install volumentations --no-build-isolation
pip uninstall importlib-metadata

cd third_party
git clone --recursive "https://github.com/NVIDIA/MinkowskiEngine"
cd MinkowskiEngine
git checkout 02fc608bea4c0549b0a7b00ca1bf15dee4a0b228
python setup.py install --force_cuda --blas=openblas
 
cd ..
cd pointnet2
python setup.py install
cd ../..

**Comment by ChunYingYu** [comment]
do you have girlfriend?

**Comment by 3406212002** [comment]
data prepare
________________________________________________________________________________________________________
let your own data be formatted in the same way as S3DIS,and put it in the right dir,then,

python -m datasets.preprocessing.s3dis_preprocessing preprocess \
    --data_dir="./data/raw/s3dis" \
    --save_dir="./data/processed/s3dis"

**Comment by 3406212002** [comment]
visualize inference outcome of your own datasets
____________________________________________________________________________________
mask sure  general.save_visualizations=true
bash scripts/s3dis/s3dis_pretrained.sh

Go to the folder where you saved the formatted file. For instance:
cd saved/area1_pretrained_eps_0.6_topk_-1_q_100/visualizations/Area_1_data_1 

Run this Python script to save a .ply file from a .bin file, for example: the result of semantic segmentation
[untitled.py](https://github.com/user-attachments/files/22609222/untitled.py)

**Comment by zhangzhenyu-zzy** [comment]
您好，可以加您微信咨询一下关于mask3d跑自己数据集的一些问题吗，我的微信或手机是18371125809，可有偿

**Comment by ArifeenDipto** [comment]
Hi, I found a way to run this Mask3D model on an upgraded  GPU version and Ubuntu.

**My GPU is:**
NVIDIA GeForce RTX 4060 (Mobile)
Architecture: Ada Lovelace (AD107)
Compute Capability: 8.9

**Driver & CUDA**
Driver Version: 590.48.01
CUDA Version (Driver-supported): 13.1

**Ubuntu version**
Ubuntu 24.04 LTS

After a lot of study I found NVIDIA GPU's are backward compatible meaning these GPU support previous CUDA versions.
For my scenario, RTX 4060 supports CUDA version up to 13.1. But the issue is this Mask3D is built using CUDA 11.3. That's why, the installation is not straightforward.

What I faced and followed to solve installation problem:

The easy way to run is through Docker. But if anyone still wants to run outside docker meaning in host machine that does not support CUDA 11.3 then they can follow my approach:

Problem 1:  During the installation of torch-scatter 2.1.2, no precompiled wheel compatible with the existing PyTorch and CUDA configuration was available, so pip downloaded the source archive (.tar.gz) and attempted to build the extension locally. The build process used the host machine’s CUDA compiler (CUDA 13.1 via nvcc) to compile the CUDA kernels. However, the installed PyTorch binary had been compiled against CUDA 11.3 and includes its own bundled CUDA runtime. As a result, the locally compiled torch-scatter extension (built with CUDA 13.1) became binary-incompatible with the PyTorch installation (built with CUDA 11.3). This CUDA version mismatch between the extension and the PyTorch runtime led to the observed runtime errors.

Solution: To resolve the CUDA version mismatch issue, I containerized the environment using Docker to ensure full consistency between the CUDA toolkit, PyTorch, and third-party extensions such as torch-scatter. I selected a Docker image with a CUDA version that matched my installed PyTorch build and compiled torch-scatter inside this controlled environment. After the compilation completed successfully, I extracted the generated wheel file from the Docker container and installed it on my host machine. This approach ensured binary compatibility with my existing PyTorch setup while avoiding conflicts with the host system’s CUDA 13.1 installation, thereby eliminating the runtime errors caused by the version mismatch.

Problem 2: Same mismatch issue with MinkowskiEngine, Pointnet2, Detectron, and Volumentation.

Solution: I followed the same approach like torch scatter for problem 1.

Problem 3:  evaluate_semantic_instance.py file used "np.int" and "np.float". 

Solution: I just removed "np" from those lines. So the remaning line had dtype=int/float

Add 8.9 here: export TORCH_CUDA_ARCH_LIST="6.0 6.1 6.2 7.0 7.2 7.5 8.0 8.6 8.9"

You can see the attached file for all the dependencies and libraries I installed for running Mask3D.

[conda_list.txt](https://github.com/user-attachments/files/25342919/conda_list.txt)

Docker setup: Ubuntu 20.04; miniconda, python 3.10 and the following file for the dependencies

[docker_conda_list.txt](https://github.com/user-attachments/files/25343311/docker_conda_list.txt)
